routes/truckRoutes.js

```diff
@@ -0,0 +1,76 @@
+// Connect to mongodb in this module as this is where you'll be making
+//creat/read/delete calls to your database
+// use 'mongodb://localhost/foodTruckAPI' for your mongoose connection string
+var express = require('express');
+var mongoose = require('mongoose');
+var foodKind;
+/*eslint-disable */
+var db = mongoose.connect('mongodb://localhost/foodTruckAPI');
+/*eslint-enable */
+var app = express();
+var router = express.Router();
+
+var bodyParser = require('body-parser');
+var urlEncoded = bodyParser.urlencoded({
+  extended: false
+});
+
+var Truck = require('../models/truckModel');
+
+app.param('type', function(request, response, next) {
+  foodKind = request.params.type;
+  var foodType = foodKind;
+  request.foodType = foodType;
+  next();
+});
+
+router.route('/')
+  .get(function(request, response) {
+    Truck.find(function(error, results) {
+      if (error) {
+        response.status(500).send(error);
+      } else {
+        response.send(results);
+      }
+    });
+  })
+  .post(urlEncoded, function(request, response) {
+    var truck = new Truck(request.body);
+
+    truck.save(function(error, book) {
+      if (error) {
+        response.status(500).send(error);
+      } else {
+        response.status(201).send(book);
+      }
+    });
+  });
+router.route('/:name')
```

Comment:
Now that we're using MongoDB, your routes should take advantage of manipulating ID's instead.